File: src/python/src/de/yaxgl/Controls/Label.py
using System;
using System.Collections.Generic;
using System.Text;

namespace de.yaxgl
{
    public class Label : Control
    {
        
        public Label(Container owner, string ID)
        {
            this.owner = owner;
            this.ID = ID;
            this.control = new System.Windows.Forms.Label();
            this.control.Click += new System.EventHandler(clickEvent);
        }

        public override void initializeNativeControl(System.Xml.XmlElement xmlElement)
        {
            setBounds(Convert.ToInt32(xmlElement.Attributes["xpos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["ypos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["width"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["height"].InnerText));
            setLabel(xmlElement.Attributes["label"].InnerText);
        }


        public string getLabel()
        {
            return this.control.Text;
        }

        public void setLabel(string label)
        {
            this.control.Text = label;
        }

    }
}

File: trunk/src/java/YaxglLibrary/de/yaxgl/Controls/Label.java
package de.yaxgl.Controls;

import org.eclipse.swt.SWT;
import org.eclipse.swt.widgets.Composite;
import org.w3c.dom.Element;

import de.yaxgl.Base.Control;
import de.yaxgl.Container.Container;

public class Label extends Control {

	public Label(Container owner,String ID)
	{
		this.owner=owner;
		this.ID=ID;	
	
		Composite composite=(org.eclipse.swt.widgets.Composite)this.owner.getNativeComponent();
		this.control=new org.eclipse.swt.widgets.Label(composite,SWT.NONE);
		
		//TODO:Label no click event in swt
		
	}
	
	@Override
	public void initializeNativeControl(Element xmlElement) {
		
		
		
		
		setBounds(Integer.valueOf(xmlElement.getAttribute("xpos")),
				Integer.valueOf(xmlElement.getAttribute("ypos")),
				Integer.valueOf(xmlElement.getAttribute("width")),
				Integer.valueOf(xmlElement.getAttribute("height")));

        setLabel(xmlElement.getAttribute("label"));
        
	}
	
	public String getLabel()
    {
        return ((org.eclipse.swt.widgets.Label)this.control).getText();
    }

    public void setLabel(String label)
    {
    	((org.eclipse.swt.widgets.Label)this.control).setText(label);
    }
}
